modules/core/shared/src/main/scala/scalacache/Cache.scala

```diff
@@ -3,19 +3,20 @@ package scalacache
 import scala.concurrent.duration.Duration
 import scala.language.higherKinds
 
-trait Cache[V] extends CacheAlg[V] {
+//todo merge with alg?
+trait Cache[F[_], V] extends CacheAlg[F, V] {
 
   def config: CacheConfig
 
   // Optimised methods for use by memoize: we know the key will be a single string so we can avoid some work.
   // These are public because calls to them are generated by the memoize macro.
 
-  def cachingForMemoize[F[_]](baseKey: String)(ttl: Option[Duration])(
+  def cachingForMemoize(baseKey: String)(ttl: Option[Duration])(
       f: => V
-  )(implicit mode: Mode[F], flags: Flags): F[V]
+  )(implicit flags: Flags): F[V]
 
-  def cachingForMemoizeF[F[_]](baseKey: String)(ttl: Option[Duration])(
-      f: => F[V]
-  )(implicit mode: Mode[F], flags: Flags): F[V]
+  def cachingForMemoizeF(baseKey: String)(ttl: Option[Duration])(
+      f: F[V]
```

Comment:
Call by name might still be useful here, since I don't think there is a restriction against using `Future` or `Try` is there?

Reply:
Neither Future nor Try have cats-effect instances, which you need anyway (Sync).